### PR TITLE
Fix/workaround for wxLocale breaking Unicode string formatting

### PR DESCRIPTION
**Pull request by lanurmi**
Also attempt to test whether wxLocale breaks formatting Unicode strings.
See #23450.

### CODE REVIEW

File: src/common/intl.cpp [hunk 1]
```diff
@@ -396,6 +396,12 @@ bool wxLocale::DoCommonPostInit(bool success,
             t->AddStdCatalog();
     }
 
+#if defined(__WXOSX__)
+    // LC_CTYPE is set in DoCommonPreInit() (see init.cpp). Set it again here,
```
Comment (vadz) [inline src/common/intl.cpp:400]: I think we might want have some `wxLocale::OSXFixCRT()` or something like this and call it from both places, even if it's a trivial one-line function because this would allow documenting this issue in a single place -- and hopefully get rid of it easier in the future, too.

**Review by vadz — COMMENTED** [review]
Thanks for the fix, we should, of course, do this if it doesn't work without it even if it's somewhat unexpected for `LC_CTYPE` to "change on its own" like this (I'd like to document it, but I'm not sure where).

I think this should also be backported to 3.2?

**Comment by lanurmi** [comment]
> Thanks for the fix, we should, of course, do this if it doesn't work without it even if it's somewhat unexpected for `LC_CTYPE` to "change on its own" like this (I'd like to document it, but I'm not sure where).

It is somewhat unexpected, though it has been changing like that already in 2.9.something. From the caller's perspective, this fix prevents it from changing again and retains old behaviour.

> I think this should also be backported to 3.2?

Absolutely, this is an issue that causes subtle bugs with random strings missing for no apparent reason or pattern.

File: src/common/intl.cpp (hunk 1 above)
Comment (vadz) [reply src/common/intl.cpp]: I've done this now, please let me know if you have any objections.
